test/built-ins/AsyncGeneratorPrototype/next/request-queue-order.js

```diff
@@ -42,22 +42,27 @@ var item1 = iter.next();
 var item2 = iter.next();
 var item3 = iter.next();
 
-var resolveorder = 0;
-
-item3.then(function(result) {
-  resolvedorder++;
-  assert.sameValue(resolvedorder, 3);
-  assert.sameValue(result.value, undefined);
-})
-
-item2.then(function(result) {
-  resolvedorder++;
-  assert.sameValue(resolvedorder, 2);
-  assert.sameValue(result.value, "second");
-})
-
-item1.then(function(result) {
-  resolvedorder++;
-  assert.sameValue(resolvedorder, 1);
-  assert.sameValue(result.value, "first");
-})
+var resolvedorder = 0;
+
+Promise.all([
+
+  item3.then(function(result) {
+    resolvedorder++;
+    assert.sameValue(resolvedorder, 3);
+    assert.sameValue(result.value, undefined);
```

Comment:
Should we also be testing the `result.done` in these?

Reply:
yes! added :)